.teamcity/_self/projects/WebApp.kt

```diff
@@ -1201,3 +1202,81 @@ object QuarantinedE2ETests: E2EBuildType(
 	buildTriggers = {
 	}
 )
+
+object DataViewsChangelogCheck : BuildType({
+	id("calypso_WebApp_DataViews_Changelog_Check")
+	uuid = "c494fae3-e0e1-49e2-9219-ea59cc17c499"
+	name = "DataViews Changelog Check"
+	description = "Ensures that changes to the packages/dataviews fork are accompanied by changes to its changelog"
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	steps {
+		bashNodeScript {
+			name = "Check DataViews changelog"
+			scriptContent = """
+				#!/usr/bin/env bash
+				set -e
+
+				# Check if there are any changes to packages/dataviews
+				DATAVIEWS_CHANGES=${'$'}(git diff --name-only refs/remotes/origin/trunk...HEAD \
+					| grep -c ^packages/dataviews/ \
+					|| true)
+
+				if [ "${'$'}DATAVIEWS_CHANGES" -eq 0 ]; then
+					exit 0
+				fi
+
+				# Check if the changelog file was modified
+				CHANGELOG_MODIFIED=${'$'}(git diff --name-only refs/remotes/origin/trunk...HEAD \
+					| grep -c '^packages/dataviews/CHANGELOG.automattic.md${'$'}' \
+					|| true)
+
+				if [ "${'$'}CHANGELOG_MODIFIED" -eq 0 ]; then
+					echo "ERROR: Changes to 'packages/dataviews' detected with no accompanying changelog entry."
+					echo "Please document your changes in 'packages/dataviews/CHANGELOG.automattic.md'."
+					exit 1
+				fi
```

Comment:
PRs that sync the package with gutenberg trunk will also fall in that bucket. Does that mean we will require a CHANGELOG entry for syncs? If yes, is that expectation clearly documented?

Reply:
Yeah, it's documented: both in the [README of the package](https://github.com/Automattic/wp-calypso/blob/trunk/packages/dataviews/README.md#contributing-to-this-package) and the corresponding FieldGuide.